game/src/js/sw.js

```diff
@@ -61,15 +43,80 @@ self.addEventListener('message', e => {
       )
     )
   }
+  if(e.data.cache) {
+    caches.open(RM_CACHE).then(function(cache) {
```

Comment:
I should have tried that myself. Thanks for the suggestions they always help me improve.

Reply:
One question though, one reason I didn't attempt this is, won't we open the cache in the case some other event happens? What happens in that case won't it throw an error because we will end up opening the cache and do nothing?

Reply:
Do you mean some other message that is not for caching? In this case, the best solution is to define some sort of message identification protocol. Maybe we should require that every message object passed into the SW has an `action` attribute that describes what it does. So here we can check `if(e.data.action == 'cache') {` before doing all the caching. Does this make sense?

Reply:
Yeah I just needed to look at it again with fresh eyes. Thanks!